dfc/tests/regression_test.go

```diff
@@ -92,6 +92,119 @@ var (
 	failLRU = ""
 )
 
+// Note: This test is most effective when multiple targets are deployed.
+func TestLocalListBucketGetTargetURL(t *testing.T) {
+	const (
+		num      = 1000
+		filesize = uint64(1024)
+		seed     = int64(111)
+		bucket   = TestLocalBucketName
+	)
+	var (
+		filenameCh = make(chan string, num)
+		errch      = make(chan error, 100)
+		sgl        *dfc.SGLIO
+		targets    = make(map[string]struct{})
+	)
+
+	if usingSG {
+		sgl = dfc.NewSGLIO(filesize)
+		defer sgl.Free()
+	}
+
+	err := client.CreateLocalBucket(proxyurl, bucket)
+	checkFatal(err, t)
+
+	defer func() {
+		err = client.DestroyLocalBucket(proxyurl, bucket)
+		checkFatal(err, t)
+	}()
+
+	putRandomFiles(0, seed, filesize, num, bucket, t, nil, errch, filenameCh, SmokeDir, SmokeStr, "", true, sgl)
+	selectErr(errch, "put", t, false)
+
+	msg := &dfc.GetMsg{GetPageSize: int(pagesize), GetTargetURL: true}
+	bl, err := client.ListBucket(proxyurl, bucket, msg, num)
+	checkFatal(err, t)
+
+	if len(bl.Entries) != num {
+		t.Errorf("Expected %d bucket list entries, found %d\n", num, len(bl.Entries))
+	}
+
+	for _, e := range bl.Entries {
+		if e.TargetURL == "" {
+			t.Error("Target URL in response is empty")
+		}
+		if _, ok := targets[e.TargetURL]; !ok {
+			targets[e.TargetURL] = struct{}{}
+		}
+		l, _, err := client.Get(e.TargetURL, bucket, e.Name, nil, nil, false, false)
+		checkFatal(err, t)
+		if uint64(l) != filesize {
+			t.Errorf("Expected filesize: %d, actual filesize: %d\n", filesize, l)
+		}
+	}
+
+	smap, err := client.GetClusterMap(proxyurl)
+	checkFatal(err, t)
+	if len(smap.Tmap) != len(targets) { // The objects should have been distributed to all targets
```

Comment:
It seems that was why you asked about even distribution.
Not sure if this check should be here. If you generate TargetURL incorrectly then direct object request to the URL should fail because rebalance is not running and target is not able to find the target inside data. 
Moreover, if you change test later to have fewer files it may fail even if everything else is good (random is random).

Reply:
It applies to local buckets only. For cloud ones I do not know a reliable way to check - direct URL request should work even if you try to download object from invalid target

Reply:
This is a good point, but I think it's a good idea to sanity check this here. Besides, the cluster map is fetched anyway for emitting a warning when only 1 target is deployed. Also, I think it's fairly safe to trust the randomness with these numbers (according to Li's earlier stats).

Reply:
OK. Since you increased the number of objects to put, it looks good to me now